### ISSUE TITLE
Lint/ShadowedArgument false positive when calling super

### ISSUE THREAD

**Issue by CyborgMaster** [rubocop-hq/rubocop#6079]
Lint/ShadowedArgument claims a variable was shadowed, when in fact it was passed to (and modified) by the superclass implementation of the method.

I believe instances like this should not be flagged.

``` ruby
  def select_fields(query, current_time)
    query = super
    ...
  end
```

--------

## Expected behavior

No offenses

## Actual behavior

```
app/models/comm_engine/media/sms_medium.rb:54:5: W: Lint/ShadowedArgument: Argument query was shadowed by a local variable before it was used.
    query = super
    ^^^^^^^^^^^^^
```

## Steps to reproduce the problem

Run the cop on a method like this:

``` ruby
  def select_fields(query, current_time)
    query = super
    query.select(*)
  end
```

## RuboCop version

```
➜  rubocop -V
0.57.2 (using Parser 2.5.1.0, running on ruby 2.3.1 x86_64-darwin16)
```

**Comment by koic** [comment]
This cop allows zero arity `super` by setting to` IgnoreImplicitReferences` configuration, so I think that this is not a bug. Default is `false` and  zero arity `super` is not allowed.

It is allowed by setting to `true` as follows.

```yaml
Lint/ShadowedArgument:
  IgnoreImplicitReferences: true
```

I think that the document is missing to `Lint/ShadowedArgument` cop about this, so I will add the document along with test cases (example code).

**Comment by CyborgMaster** [comment]
Actually, now that I think about it, I did see this option, but I figured it worked in the opposite manner.  

Wouldn't ignore `IgnoreImplicitReferences: true` imply that the cop would *ignore* the implicit reference to the call to `super` and therefore not count it as a valid use of the parameter before overwriting?  And them `IgnoreImplicitReferences: false` would meant that it wouldn't ignore them?
